Activate the Copilot button based on the available accounts and the g…

schroedermarius opened the pull request:
Closes #20698

## Description
This pull request refactors the logic for enabling the Copilot button in the `CommitMessage` component. The changes improve code readability and maintainability by encapsulating the button's enablement logic into a dedicated getter method.

Refactoring for Copilot button logic:

* [`app/src/ui/changes/commit-message.tsx`](diffhunk://#diff-d5d694641d1a8b76d94b43c41f2a588d88974d0d35906c78ae53d80385e86a97R995-R1014): Added a new getter method, `isCopilotButtonEnabled`, to encapsulate the logic for determining whether the Copilot button should be available. This replaces inline checks and simplifies conditional rendering in `renderCopilotButton`.
* [`app/src/ui/changes/commit-message.tsx`](diffhunk://#diff-d5d694641d1a8b76d94b43c41f2a588d88974d0d35906c78ae53d80385e86a97R995-R1014): Updated `isActionBarEnabled` to include the new `isCopilotButtonEnabled` method, ensuring the action bar renders correctly based on the updated logic.
* [`app/src/ui/changes/commit-message.tsx`](diffhunk://#diff-d5d694641d1a8b76d94b43c41f2a588d88974d0d35906c78ae53d80385e86a97R874-L890): Refactored `renderCopilotButton` to use the `isCopilotButtonEnabled` method, improving readability and removing redundant code.

Thanks to @tsjdev-apps to leading me into this issue.

schroedermarius committed:
Activate the Copilot button based on the available accounts and the generation function for commit messages

niik commented:
Hey @schroedermarius, thanks for wanting to contribute! The way I'm reading this it's actually not a straight up refactor, it does change the logic such that the Copilot button will now be available for non-GitHub repositories so I'd like to get @sergiou87's review on this before we continue.

If we do choose to continue we should also remove the divider when the copilot button is enabled and the co-author input isn't.

<img width="335" alt="Screenshot 2025-07-04 at 15 30 41" src="https://github.com/user-attachments/assets/2c2a5fa9-2574-4b97-833f-6ee6a21fd9b9" />

niik commented:
Ah I see now that this refers to https://github.com/desktop/desktop/issues/20698. Good, let's wait on the outcome of that issue before we move ahead

schroedermarius committed:
Show separator only when co-author input is enabled

The separator div is now conditionally rendered based on whether the co-author input is enabled, improving UI clarity.

schroedermarius commented:
* **Separator visibility**: Modified the rendering of the separator to conditionally display it only when `isCoAuthorInputEnabled` is true, ensuring cleaner UI logic.

sergiou87 reviewed (approved):
Changes make sense and work as expected, thanks for your contribution!! 💖
